### ISSUE TITLE
Add VideoPrism

### ISSUE THREAD

**Issue by MHRDYN7** [huggingface/transformers#39893]
### Model description

I'd like to contribute the VideoPrism model by google deepmind. The code base (written in flax) was made public in June and the base model weights alone has 2.7k downloads on [HF hub](https://huggingface.co/google/videoprism-base-f16r288) with almost no support except for the original repo. There are two models, a foundational video model and a video-text model, with base and large sizes.

Why add the model
* They claim that the model "achieves state-of-the-art performance on 31 out of 33 public video understanding benchmarks using a single frozen model."
* The architecture is unique with a spatial encoder layer followed by a temporal encoder layer (not fused spatio-temporal encoders) with no cls token. This architectures was introduced in the [VIVIT paper](https://arxiv.org/abs/2103.15691) but the HF VIVIT model implements only one of the four architectures proposed in that paper even though the weight for this architecture was released back then.

What the model lacks
* The VideoPrism [paper](https://arxiv.org/abs/2402.13217) was released in feb 2024, so it is old and therefore the architecture does not include modern day components like rotary positional embeddings (used in vjepa2) or GLU.
*  The foundational video model does not come with the head weights and they don't even plan to release those and that's why we'll need to train the final layer on something like the kinetics-400 dataset for classification model.

### Open source status

- [x] The model implementation is available
- [x] The model weights are available

### Provide useful links for the implementation

Paper link: https://arxiv.org/abs/2402.13217
Repo link: https://github.com/google-deepmind/videoprism
checkpoint: [video model](https://huggingface.co/google/videoprism-base-f16r288), [video-text model](https://huggingface.co/google/videoprism-lvt-base-f16r288)

@qubvel @NielsRogge I've already converted the base model's flax implementation to HF and got the exact logits for the forward pass. I'll go ahead with the rest if I get the green light for the contribution.

**Comment by MHRDYN7** [comment]
you may take a look at the implementation in the draft pr above

**Comment by NielsRogge** [comment]
Sounds good! Ideally modular can be leveraged: https://huggingface.co/docs/transformers/en/modular_transformers

**Comment by MHRDYN7** [comment]
Sure, I've used the modular approach inheriting classes from vivit.
